syntax = "proto3";

package collaboration;

// Client Service
service CollaborationService {
    rpc Login(LoginRequest) returns (LoginResponse);
    rpc Logout(LogoutRequest) returns (StatusResponse);
    rpc Post(PostRequest) returns (StatusResponse);
    rpc Get(GetRequest) returns (GetResponse);
}

// LLM Service
service LLMService {
    rpc GetLLMAnswer(LLMRequest) returns (LLMResponse);
}

// Messages
message LoginRequest {
    string username = 1;
    string password = 2;
}

message LoginResponse {
    string status = 1;
    string token = 2;
}

message LogoutRequest {
    string token = 1;
}

message StatusResponse {
    string status = 1;
    string message = 2;
}

message PostRequest {
    string token = 1;
    string type = 2;
    string data = 3;
}

message GetRequest {
    string token = 1;
    string type = 2;
    string params = 3;
}

message DataItem {
    string id = 1;
    string data = 2;
}

message GetResponse {
    string status = 1;
    repeated DataItem items = 2;
}

message LLMRequest {
    string request_id = 1;
    string query = 2;
    string context = 3;
}

message LLMResponse {
    string request_id = 1;
    string answer = 2;
}